hashes/src/sha256t.rs

```diff
@@ -62,10 +60,33 @@ impl<T: Tag> core::hash::Hash for Hash<T> {
 
 crate::internal_macros::hash_trait_impls!(256, true, T: Tag);
 
-fn from_engine<T: Tag>(e: sha256::HashEngine) -> Hash<T> {
+fn from_engine<T: Tag>(e: HashEngine) -> Hash<T> {
     use crate::Hash as _;
 
-    Hash::from_byte_array(sha256::Hash::from_engine(e).to_byte_array())
+    Hash::from_byte_array(sha256::Hash::from_engine(e.0).to_byte_array())
+}
+
+/// Engine to compute SHA256 hash function.
+#[derive(Clone)]
+pub struct HashEngine(pub(crate) sha256::HashEngine);
```

Comment:
This still leaves room for confusing differently-tagged engines. But maybe if we just tag this one and leave `sha256::HashEngine` alone it'll be less annoying for people using untagged hashes. Then we could still have `Default` as well.